### PR TITLE
emphasize luarocks "bootstrap" to avoid user confusion

### PR DESCRIPTION
**Pull request by teto**
## Description of changes

renamed the files and updated the pname

## Things done

- Built on platform(s)
  - [ ] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2411.section.md) (or backporting [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) and [24.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

File: pkgs/development/tools/misc/luarocks/default.nix [hunk 1]
```diff
@@ -66,7 +66,7 @@ stdenv.mkDerivation (finalAttrs: {
   postInstall = ''
     sed -e "1s@.*@#! ${lua}/bin/lua$LUA_SUFFIX@" -i "$out"/bin/*
     substituteInPlace $out/etc/luarocks/* \
-     --replace '${lua.luaOnBuild}' '${lua}'
```
Comment (teto) [inline pkgs/development/tools/misc/luarocks/default.nix:69]: @Artturin just wanted to confirm with you this was the correct choice and wont explode on crosscompilation

**Review by mrcjkb — APPROVED** [review]

File: pkgs/development/tools/misc/luarocks/default.nix (hunk 1 above)
Comment (arcnmx) [reply pkgs/development/tools/misc/luarocks/default.nix:69]: This seems to have caused a regression:

```bash
nix build nixpkgs#luajit_openresty.pkgs.luarocks_bootstrap
```
> substituteStream() in derivation luarocks_bootstrap-3.11.1: ERROR: pattern /nix/store/4qkdd4slrr3i8rsk0zj4kk0rmyqmbli4-luajit-2.1-20220915 doesn't match anything in file '/nix/store/g3w19f2xclqqh6yn9rf1vcnxi5icbjgx-luarocks_bootstrap-3.11.1/etc/luarocks/config-5.1.lua'
> error: builder for '/nix/store/v700wnxlzmvxsxizc86bc4q1bljcyisk-luarocks_bootstrap-3.11.1.drv' failed with exit code 1

Comment (Artturin) [reply pkgs/development/tools/misc/luarocks/default.nix:69]: Hmm that file is empty, I was wondering why it was added and found that it was added by me :D https://github.com/NixOS/nixpkgs/commit/7b24b57c14b954ff15a0b49b8a0404a222e3f258

`pkgsCross.aarch64-multiplatform.luaPackages.luarocks_bootstrap` contains 

```
-- LuaRocks configuration

rocks_trees = {
   { name = "user", root = home .. "/.luarocks" };
   { name = "system", root = "/nix/store/zsqak897x5y75i3jh9brih1pd3silpih-luarocks_bootstrap-aarch64-unknown-linux-gnu-3.11.1" };
}
variables = {
   LUA_DIR = "/nix/store/29p4fvkmbr6hy91304dcsragir7k7bmm-lua-aarch64-unknown-linux-gnu-5.2.4";
   LUA_BINDIR = "/nix/store/29p4fvkmbr6hy91304dcsragir7k7bmm-lua-aarch64-unknown-linux-gnu-5.2.4/bin";
   LUA_VERSION = "5.2";
   LUA = "/nix/store/29p4fvkmbr6hy91304dcsragir7k7bmm-lua-aarch64-unknown-linux-gnu-5.2.4/bin/lua";
}
```

Comment (Artturin) [reply pkgs/development/tools/misc/luarocks/default.nix:69]: I don't know what the conditions are to have that file populated so maybe it should be `replace-warn`?

Comment (Artturin) [reply pkgs/development/tools/misc/luarocks/default.nix:69]: https://github.com/NixOS/nixpkgs/pull/337752

**Pull request merged**